static/app/utils/profiling/frame.tsx

```diff
@@ -52,18 +49,18 @@ export class Frame extends WeightedNode {
 
     // If the frame is a web frame and there is no name associated to it, then it was likely invoked as an iife or anonymous callback as
     // most modern browser engines properly show anonymous functions when they are assigned to references (e.g. `let foo = function() {};`)
-    if (type === 'web') {
-      if (frameInfo.name === undefined || frameInfo.name === '') {
-        this.name = t('anonymous');
+    if (type === 'web' || type === 'node') {
+      if (!this.name || this.name.startsWith('unknown ')) {
```

Comment:
Right now the backend marks frames as "unknown", but after discussing with @phacops this will change and we'll fall through that first if and avoid the string operations entirely